components/Collections/tests/earlgrey/MDCCollectionViewControllerTests.m

```diff
@@ -0,0 +1,201 @@
+                                                       grey_sufficientlyVisible(),
+                                                       grey_interactable()]);
+  id<GREYMatcher> scrollableMatcher = grey_accessibilityID(scrollableIdentifier);
+  [MDCCollectionViewControllerTests scrollToElementWithMatcher:targetMatcher
+                                          inElementWithMatcher:scrollableMatcher
+                                                     direction:direction
+                                                      withStep:step
+                                                        action:tap ? grey_tap() : nil];
+}
+
+/**
+ Navigates to a specific Example screen given an array of "breadcrumb" identifiers.
+ 
+ @param breadcrumbs an array of identifiers for the example to open.
+
+ @note Only the first two elements are currently handled.
+ */
++ (void)jumpToExampleWithBreadcrumbs:(NSArray <NSString *>*)breadcrumbs {
+  NSAssert(breadcrumbs.count >= 2, @"Breadcrumb paths must be at least 2 elements.");
+  [MDCCollectionViewControllerTests scrollToElementWithIdentifier:breadcrumbs[0]
+                                          inElementWithIdentifier:@"collectionView"
+                                                        direction:kGREYDirectionDown
+                                                         withStep:-1
+                                                              tap:YES];
+  [MDCCollectionViewControllerTests
+      scrollToElementWithIdentifier:[NSString stringWithFormat:@"Cell%@", breadcrumbs[1]]
+            inElementWithIdentifier:[NSString stringWithFormat:@"Table%@", breadcrumbs[0]]
+                          direction:kGREYDirectionDown
+                           withStep:-1
+                                tap:YES];
+}
+
+/**
+ Navigates back to the main screen from the Custom Section Insets example screen.
+ */
++ (void)returnToMainScreen {
+  // Leave the Custom Section Insets view
+  [[EarlGrey selectElementWithMatcher:grey_accessibilityID(@"back_bar_button")] performAction:grey_tap()];
+  // Leave the Collections view
+  [[EarlGrey selectElementWithMatcher:grey_accessibilityID(@"back_bar_button")] performAction:grey_tap()];
+  // Scroll to the top of the main screen
+  [[EarlGrey selectElementWithMatcher:grey_accessibilityID(@"collectionView")]
+      performAction:grey_scrollToContentEdge(kGREYContentEdgeTop)];
+}
+
+/**
+ Returns a GREYActionBlock that retrieves the @c numberOfSections property value from a 
+ UICollectionView.
+ 
+ @param numberOfSections inout parameter for storing the number of sections value.
+ */
++ (GREYActionBlock *)getNumberOfSections:(NSInteger *)numberOfSections {
+  return [GREYActionBlock actionWithName:@"getNumberOfSectionsFromUICollectionView"
+                             constraints:grey_respondsToSelector(@selector(numberOfSections))
+                            performBlock:^BOOL(id element, NSError *__strong *errorOrNil) {
```

Comment:
If element will always be a UICollectionView *, then cast it right in the block declaration.